src/lib/mina/account-update.ts

```diff
@@ -897,8 +901,6 @@ class AccountUpdate implements Types.AccountUpdate {
    */
   requireSignature() {
     let { nonce, isSameAsFeePayer } = AccountUpdate.getSigningInfo(this);
-    // if this account is the same as the fee payer, we use the "full commitment" for replay protection
-    this.body.useFullCommitment = isSameAsFeePayer;
```

Comment:
previously there were two different ways we ensured that it's not replayed:
* if the account was the same as the fee payer, we used the full commitment
* otherwise, we incremented the nonce and set a nonce precondition

I like the simplification of just always using the full commitment (my initial idea on why not to do that was flexibility of signing with one key without yet having the fee payer sign; but now I think the simplicity is more important)

What this also means is that we _don't_ need to do the nonce increase in this function depending on `isSameAsFeePayer`.

Instead, I would do the following:
* add an `boolean` optional parameter to this function `noFullCommitment`, if it's true do the nonce increment instead
* remove the `isSameAsFeePayer` return value from `getSigningInfo`